src/browserWindow.ts

```diff
@@ -0,0 +1,179 @@
+import {
+    window,
+    ViewColumn,
+    WebviewPanel
+} from 'vscode';
+import { D2P } from './docToPreviewGenerator';
+
+const preText = String.raw`
+<!DOCTYPE html>
+<html lang="en" title="d2 Diagram">
+<title>Diagram</title>
+<meta name="viewport" content="width=device-width, initial-scale=1">
+<meta charset="utf-8">
+
+<head>
+	<style>
+		#toolbar {
+			z-index: 100;
+			position: fixed;
+			left: 0px;
+			top: 0px;
+			background-color: #909090;
+			padding-left: 15px;
+			padding-right: 200px;
+			padding-bottom: 5px;
+			width: 100%;
+			color: #0D32B2;
+		}
+
+		#zoomSlider {
+			appearance: none;
+			border-radius: 3px;
+			background: #0D32B2;
+			height: 5px;
+		}
+
+		#zoomSlider::-webkit-slider-thumb {
+			appearance: none;
+			box-shadow: 1px 1px 1px #000000, 0px 0px 1px #0d0d0d;
+			border: 1px solid darkblue;
+			height: 20px;
+			width: 8px;
+			margin-bottom: 5px;
+			border-radius: 3px;
+			background: lightblue;
+			cursor: pointer;
+		}
+
+		#percent {
+			display: inline-block;
+			font-family: monospace;
+			font-size: 18px;
+			font-weight: 600;
+			width: 55px;
+		}
+
+		button {
+			margin-top: 3px;
+			color: #0D32B2;
+			border-color: #0D32B2;
+			background-color: #F7F8FE;
+			border-radius: 6px;
+			border: 3 3 3 3;
+			box-shadow: 1px 1px 1px black, 0px 0px 1px 0d0d0d;
+			padding: 5px;
+			min-width: 50px;
+			font-size: 14px;
+			font-weight: 600;
+		}
+
+		svg {
+			position: absolute;
+			left: 0px;
+			top: 0px;
+			transform-origin: 0px 0px;
+			margin-top: 30px;
+		}
+	</style>
+	<div id="toolbar">
+		<button onclick="zoomMinus();" title="Zoom Out">Out</button>
+		<input id="zoomSlider" title="Zoom" onchange="sliderOnChange()" type="range" min="10" max="150" value="100"
+			step="5">
+		<button onclick="zoomPlus();" title="Zoom In">In</button>
+		<span id="percent"></span>
+		<button id="zoomFit" onclick="zoomFit()" title="Fit to window">Fit to window</button>
+	</div>
+	<script>
+		var svgElement;
+		var zoomLevel = 100;
+		var percent = document.getElementById('percent');
+		var zoomSlider = document.getElementById('zoomSlider');
+
+		function onLoad() {
+			var svgs = document.getElementsByTagName('svg');
+			svgElement = svgs[0];
+			svgElement.style.marginTop = "30px";
+			setZoomLevel(100);
```

Comment:
I think a zoomFit would be better for the default onload.

Reply:
Agree, I'll change this.

Reply:
Change [9650284](https://github.com/terrastruct/d2-vscode/pull/11/commits/9650284237a5893b5c8808b2b4bdee7254f62b04) fixes this.